classes/class-wc-connect-taxjar-integration.php

```diff
@@ -104,9 +111,92 @@ public function add_tax_settings( $tax_settings ) {
 		// Insert the "automated taxes" setting at the top (under the section title)
 		array_splice( $tax_settings, 1, 0, array( $automated_taxes ) );
 
+		if ( $this->is_enabled() ) {
+			// If the automated taxes are enabled, disable the settings that would be reverted in the original plugin
+			foreach ( $tax_settings as $index => $tax_setting ) {
+				if ( in_array( $tax_setting['id'], array( 'tax_options', 'woocommerce_tax_classes', self::OPTION_NAME ) ) ) {
+					continue;
+				}
+				$tax_settings[$index]['custom_attributes'] = array( 'disabled' => true );
+			}
+		}
+
 		return $tax_settings;
 	}
 
+	/**
+	 * When automated taxes are enabled, overwrite core tax settings that might break the API integration
+	 * This is similar to the original plugin functionality where these options were reverted on page load
+	 *
+	 * @param mixed $value - option value
+	 * @param array $option - option metadata
+	 * @return string new option value, based on the automated taxes state or $value
+	 */
+	public function sanitize_tax_option( $value, $option ) {
+		if ( ! is_array( $option ) || ! $this->is_enabled() && 'yes' != $_POST[self::OPTION_NAME] ) {
+			return $value;
+		}
+
+		switch( $option['id'] ) {
+			case 'woocommerce_calc_taxes':
+				// If automated taxes are enabled and user disables taxes we re-enable them
+				return 'yes';
+			case 'woocommerce_calc_taxes':
```

Comment:
Copy pasta? Should this be `woocommerce_tax_based_on`?

Could we reuse the `configure_tax_settings()` method somehow here instead of filtering value on save? _(Just looking to avoid duplicating this logic if we can)_